### PR TITLE
gh-98393: Update test_os for bytes-like types

### PR DESCRIPTION
**Pull request by vstinner**
Address Serhiy Storchaka's review.

* Issue: gh-98393

### CODE REVIEW

**Comment by vstinner** [comment]
cc @serhiy-storchaka

File: Lib/test/test_os.py [hunk 1]
```diff
@@ -3861,17 +3861,17 @@ def test_oserror_filename(self):
         for filenames, func, *func_args in funcs:
             for name in filenames:
                 if not isinstance(name, (str, bytes)):
-                    with self.assertRaises(TypeError):
-                        func(name, *func_args)
+                    # don't test TypeError here
```
Comment (vstinner) [inline Lib/test/test_os.py]: Other types are already well tested in other test methods of test_os.

Comment (serhiy-storchaka) [reply Lib/test/test_os.py]: Then why these filenames are occurred here at first place? `bytearray` and `memoryview` should not be added in `self.bytes_filenames`.

BTW, does checking condition `sys.platform == "win32"` above still have sense?

Comment (vstinner) [reply Lib/test/test_os.py]: I removed bytearray and memoryviews from ``self.bytes_filenames`` (and ``self.filenames``).

I removed the code specific to Windows: let's see how CIs like my updated change :-)

**Comment by vstinner** [comment]
@serhiy-storchaka: Would you mind to review the updated PR?

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by vstinner** [comment]
"Tests / Check if generated files are up to date (pull_request)" failed for an unrelated reason:

```
python3.8 ./Tools/build/update_file.py ./Include/internal/pycore_ast.h ./Include/internal/pycore_ast.h.new
  File "./Tools/build/generate_opcode_h.py", line 111
    with (open(outfile, 'w') as fobj, open(internaloutfile, 'w') as iobj):
                             ^
SyntaxError: invalid syntax
```

I rebased my PR to see if it helps.

**Comment by vstinner** [comment]
Thanks for the review @serhiy-storchaka.

**Pull request merged**